core-services/chat/src/main/java/org/silverpeas/core/chat/servers/OpenfireServer.java

```diff
@@ -0,0 +1,178 @@
+      if (response != null) {
+        response.close();
+      }
+    }
+  }
+
+  @Override
+  public void createRelationShip(final User user1, final User user2) {
+    ChatUser chatUser1 = ChatUser.fromUser(user1);
+    ChatUser chatUser2 = ChatUser.fromUser(user2);
+    Response response = null;
+    try {
+      HttpRequester requester = request(url, chatUser1.getChatLogin(), "roster");
+      response =
+          requester.post(o -> o.put("jid", chatUser2.getChatId()).put("subscriptionType", "3"));
+
+      if (response.getStatus() == HttpRequester.STATUS_CONFLICT) {
+        logger.warn("The XMPP relationship between users {0} and {1} already exists!",
+            chatUser1.getId(), chatUser2.getId());
+      } else if (response.getStatus() != HttpRequester.STATUS_CREATED) {
+        logger.error("Error while creating XMPP relationship: {0}",
+            response.getStatusInfo().getReasonPhrase());
+        throw new ChatServerException(
+            SilverpeasExceptionMessages.failureOnAdding("XMPP relationship",
+                chatUser1.getId() + " <-> " + chatUser2.getId()));
+      }
+    } catch (Exception e) {
+      logger.error("Error while creating XMPP relationship: {0}", e.getMessage());
+      throw new ChatServerException(SilverpeasExceptionMessages.failureOnAdding("XMPP relationship",
+          chatUser1.getId() + " <-> " + chatUser2.getId()), e);
+    } finally {
+      if (response != null) {
+        response.close();
+      }
+    }
+  }
+
+  @Override
+  public void deleteRelationShip(final User user1, final User user2) {
+    ChatUser chatUser1 = ChatUser.fromUser(user1);
+    ChatUser chatUser2 = ChatUser.fromUser(user2);
+
+    Response response = null;
+    try {
+      HttpRequester requester = request(chatUser1.getChatLogin(), "roster", chatUser2.getChatId());
+      response = requester.delete();
+
+      if (response.getStatus() != HttpRequester.STATUS_OK) {
+        logger.error("Error while deleting XMPP relationship: {0}",
+            response.getStatusInfo().getReasonPhrase());
+        throw new ChatServerException(
+            SilverpeasExceptionMessages.failureOnDeleting("XMPP relationship",
+                chatUser1.getId() + " <-> " + chatUser2.getId()));
+      }
+    } catch (Exception e) {
+      logger.error("Error while creating XMPP relationship: {0}", e.getMessage());
+      throw new ChatServerException(SilverpeasExceptionMessages.failureOnAdding("XMPP relationship",
+          chatUser1.getId() + " <-> " + chatUser2.getId()), e);
+    } finally {
+      if (response != null) {
+        response.close();
+      }
+    }
+  }
+
+  @Override
+  public boolean isUserExisting(final User user) {
+    ChatUser chatUser = ChatUser.fromUser(user);
+    Response response = null;
+    try {
+      HttpRequester requester = request(url, chatUser.getChatLogin());
+      response = requester.head();
+      return response.getStatus() == HttpRequester.STATUS_OK;
+
+    } catch (Exception e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1166)